jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/impl/module/FlattenedSupplierModule.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 VicTools.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.victools.jsonschema.generator.impl.module;
+
+import com.fasterxml.classmate.ResolvedType;
+import com.github.victools.jsonschema.generator.MemberScope;
+import com.github.victools.jsonschema.generator.Module;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfigBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Default module being included if {@code Option.FLATTENED_SUPPLIERS} is enabled.
+ */
+public class FlattenedSupplierModule implements Module {
```

Comment:
I'd prefer to refactor the `FlattenedOptionalModule` and extract the common generic part of unwrapping a generic's first argument instead of just adding a second specific module, which is largely identical.